src/lib/vm-manager-hoc.jsx

```diff
@@ -33,6 +33,7 @@ const vmManagerHOC = function (WrappedComponent) {
                 this.props.vm.attachAudioEngine(this.audioEngine);
                 this.props.vm.setCompatibilityMode(true);
                 this.props.vm.initialized = true;
+                this.props.vm.setLocale(this.props.locale, this.props.messages);
```

Comment:
This may be overkill, but you could check whether the locale is different before setting it. `this.props.vm.getLocale()` will return the current vm locale. If it doesn't match `this.props.locale`, then set it.

Reply:
IMO it would be better for `setLocale` to check internally if it's being asked to make a change, and early-out or otherwise do nothing if the new locale matches the old one. That way we only write that `if` statement in one place instead of in every place that calls `setLocale`.

Reply:
Yeah. That's why I'm curious about why vm's setLocale does some things conditionally, but always calls refreshBlocks.

Reply:
So checking is definitely overkill as vm does the check. There were times when the language would change and the extension blocks would end up out of sync (one language behind). I think calling refreshBlocks ensured that the extensions were always in sync. However, it does seem like we could put the refreshBlocks inside the check and not call it if the language hasn't changed.

Reply:
I did a quick check to see what would happen if the refreshBlocks wasn't called. When you switch from player to `see inside` the extension blocks are not rendered properly (red blocks). So even though vm is initialized, and the locale is set, we do need to refreshBlocks when switching between player and editor views. Basically, always calling refreshBlocks ensures that the extensions get rendered correctly regardless of the path to get here (load project with extension, add extension, see inside, etc)

Reply:
@cwillisf, @chrisgarrity, I'm curious what your thoughts are about calling `vm.setLocale()` before or after `vm.start()`? I think it makes sense, but I want to make sure because there are not too many things we call before `vm.start()`

Reply:
@kchadha are you asking about theoretical principles or concrete bugs? ;)

IMO it's morally fine to call `vm.setLocale()` before `vm.start()` in that I think of the locale as a setting that can be changed at any time. You can paint your car whether or not it's moving... in fact, some would argue it's better to paint a stationary car. Those people have no sense of adventure.

In terms of whether or not it's currently safe... uh... I'm not sure, to be honest. I would expect it to be safe and I'm not aware of anything on the VM side that would make it unsafe, but I'm less familiar with the side effects on the scratch-blocks side of things. Any thoughts on that, @chrisgarrity?